curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentEphemeralNode.java

```diff
@@ -233,8 +236,15 @@ else if ( event.getResultCode() == KeeperException.Code.OK.intValue() )
                 {
                     path = event.getName();
                 }
+                else if ( event.getResultCode() == KeeperException.Code.NOAUTH.intValue() )
+                {
+                	log.warn("Client does not have authorisation to write ephemeral node at path {}", path);
```

Comment:
It would be better to just log the bad result code. Is there a reason that the check for NOAUTH is needed? Why not just log all non-expected result codes?